src/tests/unit/tests/injected/target-page-action-message-creator.test.ts

```diff
@@ -12,67 +13,63 @@ import { EventStubFactory } from '../../common/event-stub-factory';
 
 describe('TargetPageActionMessageCreator', () => {
     const eventStubFactory = new EventStubFactory();
+    const dispatcherMock = Mock.ofType<ActionMessageDispatcher>();
     let testSubject: TargetPageActionMessageCreator;
-    let postMessageMock: IMock<(msg: Message) => void>;
-    let tabId: number;
 
     beforeEach(() => {
-        tabId = -1;
-        postMessageMock = Mock.ofInstance(message => {});
-        testSubject = new TargetPageActionMessageCreator(postMessageMock.object, tabId, new TelemetryDataFactory());
+        dispatcherMock.reset();
+        testSubject = new TargetPageActionMessageCreator(new TelemetryDataFactory(), dispatcherMock.object);
     });
 
-    test('scrollRequested', () => {
-        const expectedMessage = {
+    it('dispatches message for scrollRequested', () => {
+        const expectedMessage: Message = {
             messageType: Messages.Visualizations.Common.ScrollRequested,
         };
 
-        postMessageMock.setup(pm => pm(It.isValue(expectedMessage))).verifiable();
-
         testSubject.scrollRequested();
 
-        postMessageMock.verifyAll();
+        dispatcherMock.verify(dispatcher => dispatcher.dispatchMessage(expectedMessage), Times.once());
     });
 
-    test('openIssuesDialog', () => {
-        const payload = {
-            eventName: TelemetryEvents.ISSUES_DIALOG_OPENED,
-            telemetry: {
-                source: TelemetryEventSource.TargetPage,
-                triggeredBy: 'N/A',
-            },
-        };
-
-        const expectedMessage = {
-            messageType: Messages.Telemetry.Send,
-            tabId: tabId,
-            payload: payload,
+    it('sends telemetry for openIssuesDialog', () => {
+        const eventName = TelemetryEvents.ISSUES_DIALOG_OPENED;
+        const eventData: TelemetryEvents.TelemetryData = {
+            source: TelemetryEventSource.TargetPage,
+            triggeredBy: 'N/A',
         };
 
-        postMessageMock.setup(pm => pm(It.isValue(expectedMessage))).verifiable();
-
         testSubject.openIssuesDialog();
 
-        postMessageMock.verifyAll();
+        dispatcherMock.verify(dispatcher => dispatcher.sendTelemetry(eventName, eventData), Times.once());
     });
 
-    test('setHoveredOverSelector', () => {
+    it('dispatches message for setHoveredOverSelector', () => {
         const selector = ['some selector'];
 
         const expectedMessage = {
             messageType: Messages.Inspect.SetHoveredOverSelector,
-            tabId: tabId,
             payload: selector,
         };
 
-        postMessageMock.setup(pm => pm(It.isValue(expectedMessage))).verifiable();
-
         testSubject.setHoveredOverSelector(selector);
 
-        postMessageMock.verifyAll();
+        dispatcherMock.verify(dispatcher => dispatcher.dispatchMessage(expectedMessage), Times.once());
     });
 
-    test('openSettingsPanel', () => {
+    it('sends telemetry for copyIssueDetailsClicked', () => {
```

Comment:
Was this previously not tested?

Reply:
It was not. There were originally 4 total unit tests (while there are 5 functions) and code coverage for this file was < 100%.